OpenRA.Mods.Common/Traits/RevealOnFire.cs

```diff
@@ -0,0 +1,79 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Reveal this actor to the target's owner when attacking.")]
+	public class RevealOnFireInfo : ConditionalTraitInfo
+	{
+		[Desc("The armament types which trigger revealing.")]
+		public readonly string[] ArmamentNames = { "primary", "secondary" };
+
+		[Desc("Stances related to the target player this actor will be revealed to during firing.")]
+		public readonly Stance TargetPlayerStances = Stance.Ally;
```

Comment:
The naming and description here are a bit hard to understand.  Based just on the name `TargetPlayerStances` sounds like it would be the stances *that this trait triggers on* rather than defining who else to reveal to.   I don't have any suggestions for a better name.